cmd/agent/subcommands/snmp/command.go

```diff
@@ -223,59 +225,23 @@ func snmpwalk(config config.Component, cliParams *cliParams) error {
 	}
 
 	// Set the snmp version
-	if cliParams.snmpVersion == "1" {
-		setVersion = gosnmp.Version1
-	} else if cliParams.snmpVersion == "2c" || (cliParams.snmpVersion == "" && cliParams.communityString != "") {
-		setVersion = gosnmp.Version2c
-	} else if cliParams.snmpVersion == "3" || (cliParams.snmpVersion == "" && cliParams.user != "") {
-		setVersion = gosnmp.Version3
-	} else {
+	if !snmp.CheckAndSetVersion(cliParams.snmpVersion, &setVersion, cliParams.communityString, cliParams.user) {
 		fmt.Printf("SNMP version not supported: %s, using default version 2c. \n", cliParams.snmpVersion) // match default version of the core check
 		setVersion = gosnmp.Version2c
 	}
 
 	// Set v3 security parameters
 	if setVersion == gosnmp.Version3 {
 		// Authentication Protocol
-		switch strings.ToLower(cliParams.authProt) {
-		case "":
-			authProtocol = gosnmp.NoAuth
-		case "md5":
-			authProtocol = gosnmp.MD5
-		case "sha":
-			authProtocol = gosnmp.SHA
-		case "sha224", "sha-224":
-			authProtocol = gosnmp.SHA224
-		case "sha256", "sha-256":
-			authProtocol = gosnmp.SHA256
-		case "sha384", "sha-384":
-			authProtocol = gosnmp.SHA384
-		case "sha512", "sha-512":
-			authProtocol = gosnmp.SHA512
-		default:
+		if !gosnmplib.SwitchAuthProtocol(cliParams.authProt, &authProtocol) {
```

Comment:
It's a interesting way to set the value via pointer, but IMHO it's not very idiomatic in go.

Maybe we can use `GetAuthProtocol()` instead with handling of the returned error.

That's said, I think we should only scope this PR to handle the default protocols for corecheck and leave other changes for a later or separate PR.